minecraft/1.11/src/main/java/nova/worldgen/wrapper/mc/forge/v1_11/wrapper/world/forward/FWWorldGenerator.java

```diff
@@ -74,24 +76,18 @@ public void generate(Random random, int chunkX, int chunkZ, World world, IChunkG
 				//  DEEPERER = 20 (15-25)
 				//  REALLYDEEP = 10 (5-15)
 
-				if (ore.oreLayers.allows(OreHeight.SURFACE))
-					oreHeightMap.put(j++, OreHeight.SURFACE);
-				if (ore.oreLayers.allows(OreHeight.UNDERSURFACE))
-					oreHeightMap.put(j++, OreHeight.UNDERSURFACE);
-				if (ore.oreLayers.allows(OreHeight.DEEP))
-					oreHeightMap.put(j++, OreHeight.DEEP);
-				if (ore.oreLayers.allows(OreHeight.DEEPER))
-					oreHeightMap.put(j++, OreHeight.DEEPER);
-				if (ore.oreLayers.allows(OreHeight.DEEPERER))
-					oreHeightMap.put(j++, OreHeight.DEEPERER);
-				if (ore.oreLayers.allows(OreHeight.REALLYDEEP))
-					oreHeightMap.put(j++, OreHeight.REALLYDEEP);
+				Iterator<OreHeight> iterator = Arrays.stream(OreHeight.values()).filter(ore.oreLayers::allows).iterator();
+
+				for (OreHeight height = iterator.next(); iterator.hasNext();) {
```

Comment:
Fixed now.
The main reason I did it that way initially, is because I made use of the external variable `j`, which must be final or effectively final to be used in lambda expressions.